### DISCUSSION TITLE
Correct way to install heroku executable?

### DISCUSSION THREAD

**Discussion by mustaqimM** [zdharma-continuum/zinit#365] [Q&A] [answered]
Hey there, I currently have heroku set up like so:
```zsh
zi lucid light-mode wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]' extract"!-" id-as"heroku" \
  nocompile sbin'*bin/heroku->heroku' for \
  "https://cli-assets.heroku.com/heroku-linux-x64.tar.gz"

zi lucid ice wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]'
zi snippet "OMZ::plugins/heroku/heroku.plugin.zsh"
```
But when I load it, I get this:
![image](https://user-images.githubusercontent.com/24436622/186203270-2c1b5aaf-8892-40a9-a5a5-db4dbd7842dd.png)

It works fine except for it showing `Snippet not loaded (heroku)`

---

~Also is there a zinit way to rename/delete a file in the directory? Annoyingly, heroku first chooses the vendored node exe in the dir before a system or env node.~
`mv"bin/node -> bin/node.BAK"` can be used with `sbin`

**Comment by vladdoster** [comment]
I can't tell if you solved this or not. Regardless, the snippet you provided could be improved.

> ```shell
> zi lucid light-mode wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]' extract"!-" id-as"heroku" \
>   nocompile sbin'*bin/heroku->heroku' for \
>   "https://cli-assets.heroku.com/heroku-linux-x64.tar.gz"
> 
> zi lucid ice wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]'
> zi snippet "OMZ::plugins/heroku/heroku.plugin.zsh"
> ```

---

What is your reasoning for the ice?

```zsh
wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]'
```

**Reply by mustaqimM** [reply]
> What is your reasoning for the ice?
> 
> ```shell
> wait'[[ -n ${ZLAST_COMMANDS[(r)her*]} ]]'
> ```

It's so it's only loaded when I start typing `her<plugin now loaded>`, I saw it in zinit's previous repo. (Is that what you're asking?)
I don't use the command often, so it's better to load it when I actually need it.

How could it be improved? It works but after typing `her`, I get that `Snippet not loaded (heroku)` message which is easily cleared. It works so I don't mind it.

**Comment by vladdoster** [comment, marked as answer]
@mustaqimM, 

It is not currently possible to use official Heroku URLs.

The issue with Heroku URLs is that (AFAIK) are AWS S3 buckets with a public access policy that does not allow listing resources. In other words, you must know the **FULL** URL otherwise the server responds with an error.

Due to the aforementioned access policy, Zinit can not get a list of URLs to parse.

---

Apologies for the delayed response.
